#' @title rbiosnp_readGVCF
#'
#' @description The function loads a \code{vcf} file resulted from the GATK suite.
#' @param file Input \code{vcf} file.
#' @return A \code{gvcf} object containing the information from the VCF file.
#' @details This function deals with the \code{GVCF} primarily generated by the HaplotypeCaller command.
#'          GVCF stands for Genomic VCF. A GVCF is a kind of VCF;
#'          so the basic format specification is the same as for a regular VCF (see the spec documentation here),
#'          but a Genomic VCF contains extra information.
#'          The key difference between a regular VCF and a GVCF is that the GVCF has records for all sites,
#'          whether there is a variant call there or not.
#' @examples
#' \dontrun{
#' cohort_gvcf <- rbiosnp_readGVCF(file = "rawSNP.g.vcf")
#' }
#' @export
rbiomirgs_gmt <- function(file){
  # f <- file
  # class(f) <- "gvcf"
  # return(f)
}

#' @title rbiosnp_qcGVCF
#'
#' @description The function extracts QC metrics from the \code{gvcf} object.
#' @param object Input \code{gvcf} file.
#' @return A \code{gvcf_qc} object containing the QC information from the input \code{GVCF} file.
#' @details This QC metrics extracted include: QD, MQ, FS, SOR, MQRnakSUM and ReadPosRankSum.
#'         QD: QualByDepth.
#'             This is the variant confidence (from the QUAL field) divided by the unfiltered depth of non-hom-ref samples.
#'             This annotation is intended to normalize the variant quality in order to avoid inflation caused when there is deep coverage.
#'             For filtering purposes, it is better to use QD than either QUAL or DP directly.
#'
#'         MQ: RMSMappingQuality.
#'             This is the root mean square mapping quality over all the reads at the site.
#'             Instead of the average mapping quality of the site, this annotation gives the
#'             square root of the average of the squares of the mapping qualities at the site.
#'             It is meant to include the standard deviation of the mapping qualities.
#'             Including the standard deviation allows us to include the variation in the dataset.
#'             A low standard deviation means the values are all close to the mean,
#'             whereas a high standard deviation means the values are all far from the mean.
#'             When the mapping qualities are good at a site, the MQ will be around 60.
#'
#'         FS: FisherStrand.
#'             This is the Phred-scaled probability that there is strand bias at the site.
#'             Strand Bias tells us whether the alternate allele was seen more or less often on the
#'             forward or reverse strand than the reference allele. When there little to no strand
#'             bias at the site, the FS value will be close to 0.
#'
#'         SOR: StrandOddsRatio.
#'              This is another way to estimate strand bias using a test similar to the symmetric odds ratio test.
#'              SOR was created because FS tends to penalize variants that occur at the ends of exons.
#'              Reads at the ends of exons tend to only be covered by reads in one direction and FS gives those variants a bad score.
#'              SOR will take into account the ratios of reads that cover both alleles.
#'              Notice most variants have an SOR value less than 3, and almost all variants have an SOR value less than 9.
#'
#'         MQRankSum: MappingQualityRankSumTest.
#'                    This is the u-based z-approximation from the Rank Sum Test for mapping qualities.
#'                    It compares the mapping qualities of the reads supporting the reference allele and the alternate allele.
#'                    A positive value means the mapping qualities of the reads supporting the alternate allele are higher than
#'                    those supporting the reference allele; a negative value indicates the mapping qualities of the reference
#'                    allele are higher than those supporting the alternate allele.
#'                    A value close to zero is best and indicates little difference between the mapping qualities.
#'
#'         ReadPosRankSum: ReadPosRankSumTest.
#'                         This is the u-based z-approximation from the Rank Sum Test for site position within reads.
#'                         It compares whether the positions of the reference and alternate alleles are different within the reads.
#'                         Seeing an allele only near the ends of reads is indicative of error, because that is where sequencers tend to make the most errors.
#'                         A negative value indicates that the alternate allele is found at the ends of reads more often than the reference allele;
#'                         a positive value indicates that the reference allele is found at the ends of reads more often than the alternate allele.
#'                         A value close to zero is best because it indicates there is little difference between the positions of the reference and alternate alleles in the reads.
#' @examples
#' \dontrun{
#' cohort_gvcf <- rbiosnp_readGVCF(file = "cohort.g.vcf")
#' gvcf_qc_metrics <- rbiosnp_qcGVCF(object = cohort_gvcf)
#' }
#' @export
rbiosnp_qcGVCF <- function(object){
  # out <- object
  # class(out) <- "gvcf_qc"
  # return(out)
}
